cpp/src/Ice/SSL/SchannelTransceiverI.cpp

```diff
@@ -63,6 +63,32 @@ namespace
         }
         return 0;
     }
+
+    HCERTCHAINENGINE defaultChainEngine(HCERTSTORE trustedRootCertificates)
+    {
+        if (trustedRootCertificates)
+        {
+            // Create a chain engine that uses our Trusted Root Store
+            CERT_CHAIN_ENGINE_CONFIG config;
+            memset(&config, 0, sizeof(CERT_CHAIN_ENGINE_CONFIG));
+            config.cbSize = sizeof(CERT_CHAIN_ENGINE_CONFIG);
+            config.hExclusiveRoot = trustedRootCertificates;
+
+            HCERTCHAINENGINE chainEngine;
+            if (!CertCreateCertificateChainEngine(&config, &chainEngine))
+            {
+                throw InitializationException(
+                    __FILE__,
+                    __LINE__,
+                    "SSL transport: error creating certificate chain engine:\n" + IceUtilInternal::lastErrorToString());
+            }
+            return chainEngine;
+        }
+        else
+        {
+            return HCCE_CURRENT_USER;
```

Comment:
We can also add a defaultChainEngine, for cases where the user needs to use HCCE_LOCAL_MACHINE. With the current API you would have to provide your validation callback and create the chain engine. You would typically want to use HCCE_LOCAL_MACHINE when running a server under a system account rather than a user account.